lib/index.js

```diff
@@ -273,46 +275,30 @@ module.exports = class DockerComposeParser {
   }
 
   /**
-   * Adds a main service if one doesn't already exist
-   *
-   * @param {Object<String:Service>} services - Object with service names as key
-   * @param {String}                 repoName - name of the repo
-   * @private
-   */
-  static _checkIfMissingMain (services, repoName) {
-    const mainServiceName = DockerComposeParser._getMainName(services)
-    if (!mainServiceName) {
-      services[repoName] = {
-        build: {
-          disabled: true
-        },
-        image: 'busybox'
-      }
-    }
-  }
-
-  /**
-   * Adds a main service if one doesn't already exist
+   * Adds a main service if one doesn't already exist to the mains.builds object, and adds it to the services array
    *
-   * @param {Array<Service>} services - Array of services
-   * @param {Object} opts
-   * @param {String} opts.repositoryName - Name of repository in GH or other service
-   * @param {String} opts.userContentDomain - User content domain for API environment. Used for creating hostnames.
-   * @param {String} opts.ownerUsername - Organization name in Runnable. Used for creating hostnames.
-   * @param {String} opts.scmDomain - Domain for used SCM. E.g. github.com
+   * @param {Service[]} services                  - Array of services
+   * @param {Object}    mains                     - Object containing all of the main builds and externals
+   * @param {Object}    mains.externals           - All services with external (github) builds (indexed by name)
+   * @param {Object}    mains.builds              - All services with local builds (same repo as compose file) (indexed by name)
+   * @param {Object}    opts
+   * @param {String}    opts.repositoryName       - Name of repository in GH or other service
+   * @param {String}    opts.userContentDomain    - User content domain for API environment. Used for creating hostnames.
+   * @param {String}    opts.ownerUsername        - Organization name in Runnable. Used for creating hostnames.
+   * @param {String}    opts.scmDomain            - Domain for used SCM. E.g. github.com
    * @private
    */
-  static _checkIfMissingMainToArray (services, opts) {
-    const repositoryName = opts.repositoryName
-    const userContentDomain = opts.userContentDomain
-    const ownerUsername = opts.ownerUsername
-    const mainService = services.find(service => service.metadata.isMain)
-    if (!mainService) {
+  static addMainIfMissing (services, mains, opts) {
+    if (!Object.keys(mains.builds).length && !Object.keys(mains.externals).length) {
+      const repositoryName = opts.repositoryName
+      const userContentDomain = opts.userContentDomain
+      const ownerUsername = opts.ownerUsername
       const newMainService = {
         metadata: {
           name: repositoryName,
           isMain: true,
-          envFiles: []
+          envFiles: [],
+          isMissingMaster: true
```

Comment:
it's a way to signify that this instance was created by default since there were no masters